doc/develop/test/twister.rst

```diff
@@ -823,6 +823,34 @@ line break instead of white spaces.
 
 Most everyday users will run with no arguments.
 
+Selecting platform scope:
+*************************
+
+One of the key features of Twister is its ability to decide on which platforms a given
+test scenario should run. This behavior has its roots in Twister being developed as
+a test runner for Zephyr's CI. With hundreds of available platforms and thousands of
+tests, the testing tools should be able to adapt the scope and select/filter out what
+is relevant and what is not.
+
+Twister is always preparing an initial list of platforms in scope for a given test,
+based on command line arguments and the test's configuration file. Then, platforms
+that don't fulfill the required conditions are filtered out from the scope.
+Using ``--force-platform`` turns off filtering.
+
+Command line arguments define the initial scope in the following way:
+
+* ``-p/--platform <platform_name>`` (can be used multiple times): only platforms
+  passed with this argument;
+* ``--all``: all available platforms;
+* ``-G/--integration``: all platforms from an `integration_platforms` list in
+  a given test configuration file. If a test has no `integration_platforms`
+  the "scope presumption"* will happen;
+* No scope argument: `"scope presumption"` will happen.
+
+`"Scope presumption"`: A list of Twister's default platforms is used as
+the initial list. If nothing is left after the filtration, a platform_allow list
+is used as the initial scope.
```

Comment:
Looks good, but could you expand some more on this section to explain the behavior as @theob-pro was trying to do in https://github.com/zephyrproject-rtos/zephyr/pull/75699?

Reply:
Yes please, you describe the command line argument but could you also describe the keys in the test case file and the "default" testing mechanism of boards.

Reply:
> Yes please, you describe the command line argument but could you also describe the keys in the test case file and the "default" testing mechanism of boards.

Those are already part of the documentation. I added links to relevant sections.

> could you expand some more on this section to explain the behavior as @theob-pro was trying to do in https://github.com/zephyrproject-rtos/zephyr/pull/75699?

Sorry, but no. As in other comment: I believe all logic required to answer this is presented in the PR. I tried to explain multiple times (it is also already part of the docs at https://docs.zephyrproject.org/latest/develop/test/twister.html#tests at `platform_allow`) that platform_allow shouldn't be treated as a scope defining parameter but a filter. Therefore, I am against describing this specific combinations since IMO it is legitimizing misuse of platform_allow.